static/src/javascripts/bootstraps/enhanced/notifications.js

```diff
@@ -43,37 +48,65 @@ define([
+                    subscribed: subscribed,
+                    icon: svgs(subscribed ? 'notificationsOff' : 'notificationsOn')
                 });
 
+            $follow.html(src);
+            bean.one($follow[0], 'click', '.js-notifications__button', handler );
+
+        },
+
+        showExplainer: function() {
+            var src = template(explainer,{
+                closeIcon : svgs('closeCentralIcon'),
+                imgMobile: svgs('notificationsExplainerMobile', ['mobile-only', 'live-notifications-explainer-svg']),
+                imgDesktop: svgs('notificationsExplainerDesktop', ['hide-on-mobile', 'live-notifications-explainer-svg'])
+            });
+
             fastdom.write(function () {
-                $('.js-notifications').prepend(src);
-                bean.one(document.body, 'click', '.js-notifications-subscribe-link', handler);
+                var notifications = $('.js-live-notifications');
+                notifications.append(src);
+                bean.one(notifications[0], 'click', '.js-live-notifications__item__close', function() {
+                    fastdom.write(function() {
+                        userPrefs.set(explainerDismissed, true);
+                        $('.js-live-notifications-explainer').remove();
+                    });
+                });
             });
         },
 
-        setSubscriptionStatus: function (subscribed) {
-            var subscribeButton = $('.js-notifications-subscribe-link'),
-                subscribeLink = $('.notifications-subscribe-link--follow'),
-                handler = subscribed ? modules.unSubscribeHandler : modules.subscribeHandler;
-            if (subscribed) {
-                subscribeLink.addClass('notifications-subscribe-link--has-subscriptions');
-            }
+        closeDisplayMessage: function(){
+            $('.js-live-notifications-denied').remove();
+            bean.one($('.js-live-notifications')[0], 'click', '.js-notifications__button', modules.subscribeHandler);
+        },
 
-            subscribeButton[0].textContent = subscribed ?  'Unfollow' : 'Follow story';
-            bean.one(document.body, 'click', '.js-notifications-subscribe-link', handler);
+        notificationsDeniedMessage: function() {
+            var src = template(permissionsTemplate,{closeIcon : svgs('closeCentralIcon')});
+            fastdom.write(function () {
+                var blocked = $('.js-notificications-blocked');
+                blocked.prepend(src);
+                bean.one(blocked[0], 'click', '.js-live-notifications-denied__item__close', modules.closeDisplayMessage);
+            });
         },
 
         subscribeHandler: function () {
-            modules.subscribe().then(modules.follow);
+            modules.subscribe().then(modules.follow)
+                .catch( function () {
+                    if (Notification.permission === 'denied') {
+                        modules.notificationsDeniedMessage();
+                    } });
```

Comment:
Here we are catching all rejected promises and then checking if permissions for notifications have been denied.

What if the error is not related to permissions? It would silently fail? Could you instead look at the type of error that is coming through, and use that to decide whether it is a permissions error? If it's not a permissions error you can rethrow the error (as you don't want to handle it here).